rpcs3/Emu/RSX/GSRender.cpp

```diff
@@ -25,12 +25,21 @@ GSRender::~GSRender()
 	}
 }
 
-void GSRender::on_init_rsx()
+bool GSRender::set_window_visible(bool visible)
 {
-	if (m_frame)
+	if (!m_frame)
 	{
-		m_frame->show();
+		return false;
 	}
+
+	if (visible != std::exchange(m_window_visible, visible))
```

Comment:
this function is so overly chaotic and horrible to read as a normal person ...
Just write a function is_window_visible() and use it where you currently aquire the return value of this function.

Also, the m_window_visible part does not make any sense, as we only want to keep the window hidden in specific circumstances, but hide it unconditionally...

Reply:
"as a normal person"? Wtf is up with the hostility Mega. That's not necessary.

Reply:
you're mistaken.
a "normal person" aka some dude who reads this for the first time.
It's completely unnecessary to write a setter that does 3 different things and acts as a getter at the same time, when you can simply add a line to check that state first and everyone gets what's happening without having to study the functions return value.
KISS

Reply:
I understood the intent... I'm talking about how you phrased it as a personal attack for no reason whatsoever.

Reply:
I guess you can interpret anything into the things I say.

Reply:
Whatever the intention is this is not a helpful statement and may be misinterpreted by other people.

Reply:
"A normal person" clearly implies elad is "not normal" in my view, but whatever.

I'm sure you can stand up for yourself if you want to.

Reply:
Who's normal in this project?

Reply:
lol